client/state/posts/actions.js

```diff
@@ -68,3 +71,36 @@ export function requestSitePosts( siteId, query = {} ) {
 		} );
 	};
 }
+
+/**
+ * Triggers a network request to fetch a specific post from a site.
+ *
+ * @param  {Number}   siteId Site ID
+ * @param  {Number}   postId Post ID
+ * @return {Function}        Action thunk
+ */
+export function requestSitePost( siteId, postId ) {
+	return ( dispatch ) => {
+		dispatch( {
+			type: POST_REQUEST,
+			siteId,
+			postId
+		} );
+
+		return wpcom.site( siteId ).post( postId ).get().then( ( post ) => {
+			dispatch( receivePost( post ) );
+			dispatch( {
+				type: POST_REQUEST_SUCCESS,
+				siteId,
+				postId
+			} );
+		} ).catch( ( error ) => {
```

Comment:
>Why use `catch` instead of passing the error handler as the second arg to `then`?

Don't feel strongly either way, but I suppose an argument could be made that it lends more clarity to what the code is effectively accomplishing, as `catch` is more meaningful than happening to be passed as the second argument.